classes/asserter/resolver.php

```diff
@@ -50,13 +52,28 @@ public function resolve($asserter)
 		{
 			$class = $this->checkClass($asserter);
 		}
-		else foreach ($this->namespaces as $namespace)
+		else
 		{
-			$class = $this->checkClass($namespace . '\\' . $asserter);
+			if (false === test::isValidIdentifier($asserter))
+			{
+				return null;
```

Comment:
AFAIR we did not run this kind of check before but you are right, with the actual code (before your patch) if something was wrong (asserter not found) we just returned `null`. I'm wondering if this is good.

It does not make sense to return `null` for an invalid identifier: it's not valid so there will never be any asserter class matching it. Requesting an asserter with an invalid identifier is just wrong IMHO.

But now I have another question: why not moving the `isValidIdentifier` to the very beginning of the `resolve` method ?